### ISSUE TITLE
Upload progress looks weird

### ISSUE THREAD

**Issue by andrey-str** [haiwen/seafile-client#751]
Hi,

We noticed that when you upload a big folder with many subfolders(in my tests I used brew's Cellar directory) synchronisation progress goes from 0% to 100% several times. Is it a bug or by design? What is the meaning of this?
We made a fork of seafile-client for our service and would like to change this behaviour to more 'natural': like going from 0% to 100% only once :-) Could you give any hint how to achieve this, were to look, or maybe this is impossible due some limitations? Appreciate your answer a lot. Thanks!

Our project is here: http://renerlion.com

**Comment by MarcoWue** [comment]
This may be related to this issue:
https://github.com/haiwen/seafile-client/issues/717

**Comment by shoeper** [comment]
This is due to the way how Seafile upload works. So multiple times from 0 to 100 is normal (as long as more than x MB did change (either 100 or 300 if I remember right).

**Comment by andrey-str** [comment]
@shoeper is this due how does Seafile-*client* uploads work or it is related to server's logic? Also, any idea, can this x MB size be changed to some value (INFINITY, for example)?

**Comment by shoeper** [comment]
Changing this size to infinity does not make sense.

Seafile works like git. The client locally computes blocks and uses CDC. This is done in batches and while the next batch is being computed the last batch(es) is getting uploaded.
Changing to inifity would first index everything and upload everything next. Additionally it would require much local disk space and could lead to running out of space.

**Comment by shoeper** [comment]
duplicate of #60
